viewing_party/main.py

```diff
@@ -0,0 +1,172 @@
+    # sorted puts user_genres' keys in a list ordered from lowest to highest value
+    # ...so ... genre_sort = ["Fantasy", "Intrigue"] 
+    popular_genre = genre_sort[-1]
+    # popular_genre = Intrigue
+    return popular_genre
+
+###################### WAVE 3 PASSED #########################
+### Wave 3.1 (2 tests) PASSED ###
+# this solution uses sets to modify the data
+# ...specializes function for removing data that's irrelevant
+def unique_helper(user_data):
+    user_set = set()
+    friend_set = set()
+
+    for user_movies in user_data["watched"]:
+        user_set.add(user_movies["title"]) 
+    for friend_items in user_data["friends"]:
+        for movie in friend_items["watched"]:
+            for title,val in movie.items():
+                friend_set.add(val)
+    return user_set, friend_set
+
+def get_unique_watched(user_data):
+    # Return a list of dictionaries that represents list of movies
+    # extra step to put the data back into a dictionary -refactor
+    user_set, friend_set = unique_helper(user_data)
+    unique2user = (user_set - friend_set)
+    # unique2user = {'Title B', 'Title E'}
+    users_unique_list = []
+    for i in unique2user:
+        unique_user_movies = {"title": None}
+        unique_user_movies["title"] = i
+        users_unique_list.append(unique_user_movies)
+    # users_unique_list = [{'title': 'Title B'}, {'title': 'Title E'}]
+    return users_unique_list
+
+### Wave 3.2 (3 tests) PASSED ###
+def get_friends_unique_watched(user_data):
+    user_set, friend_set = unique_helper(user_data)
+    # Determine which movies friends have watched, but the user has not watched.
+    unique2friends = (friend_set - user_set)
+    friends_unique_list = [] 
+    for i in unique2friends:
+        unique_friends_movies = {"title": None}
+        unique_friends_movies["title"] = i
+        friends_unique_list.append(unique_friends_movies)
+    return friends_unique_list
+
+###################### WAVE 4 PASSED #########################
+### Wave 4.1 (2 tests) PASSED ###
+def get_available_recs(user_data):
+    recommendations = []
+    user_subscriptions = user_data["subscriptions"]
+
+    for collection in user_data["friends"]:
+        for movie in collection["watched"]:
+            if movie in recommendations:
+                continue
+            elif movie not in user_data["watched"] and movie["host"] in user_subscriptions:
+                recommendations.append(movie)
+    return recommendations
+                
+
+###################### WAVE 5 #########################
+### Wave 5.1 (4 tests) PASSED ###
+def get_new_rec_by_genre(user_data):
+    recommend_by_genre = []
+
+    popular_genre = get_most_watched_genre(user_data)
+    # Intrigue
+
+    if user_data["watched"] == []:
+        recommend_by_genre = user_data["watched"]
+    else:   
+        for account in user_data["friends"]:
+            for movie in account["watched"]:
+                if popular_genre in movie["genre"]:
```

Comment:
Instead of doing `in` here, I'd just check to see if they are equal.  

```suggestion
                if popular_genre == movie["genre"]:
```

Reply:
Nice! I failed with the == approach and it might have been a syntax error and I just threw the baby out with the bath. is the == better because the "in" is a loop and it's already looping through the values?